borg/cache.py

```diff
@@ -390,6 +400,39 @@ def legacy_cleanup():
             self.do_cache = os.path.isdir(archive_path)
             self.chunks = create_master_idx(self.chunks)
 
+    def check_cache_compatibility(self):
+        my_features = Manifest.SUPPORTED_REPO_FEATURES
+        if self.ignored_features & my_features:
+            return False
```

Comment:
would be cool if the hard to understand stuff ends up in some helpful comments in the src. :)